src/main/kotlin/com/bugsnag/android/gradle/internal/DexguardCompat.kt

```diff
@@ -0,0 +1,85 @@
+package com.bugsnag.android.gradle.internal
+
+import com.android.build.gradle.api.ApkVariant
+import com.android.build.gradle.api.ApkVariantOutput
+import com.bugsnag.android.gradle.GroovyCompat
+import org.gradle.api.Project
+import java.io.File
+import java.nio.file.Paths
+
+/**
+ * Finds the mapping file locations for DexGuard >=9. This can be different depending on whether it
+ * is a bundle or an APK.
+ */
+internal fun findMappingFileDexguard9(
+    project: Project,
+    variant: ApkVariant,
+    variantOutput: ApkVariantOutput
+): List<File> {
+    return listOf(
+        findDexguardMappingFile(project, variant, variantOutput, "outputs", "dexguard", "mapping", "apk"),
+        findDexguardMappingFile(project, variant, variantOutput, "outputs", "dexguard", "mapping", "bundle")
+    )
+}
+
+/**
+ * Finds the mapping file location for DexGuard <9
+ */
+internal fun findMappingFileDexguardLegacy(
+    project: Project,
+    variant: ApkVariant,
+    variantOutput: ApkVariantOutput
+): File {
+    return findDexguardMappingFile(project, variant, variantOutput, "outputs", "mapping")
+}
+
+/**
+ * Retrieves the location of a DexGuard mapping file for the given variantOutput.
+ * The expected location for this is:
+ * /build/outputs/mapping/<productFlavor>/<buildType>/<split>/mapping.txt
+ *
+ * variant.mappingFile cannot currently be overridden using the AGP DSL on a per variantOutput
+ * basis, which is why the DexGuard plugin sets a different output for its mapping files.
+ * see https://issuetracker.google.com/issues/78921539
+ */
+private fun findDexguardMappingFile(
+    project: Project,
+    variant: ApkVariant,
+    variantOutput: ApkVariantOutput,
+    vararg path: String
+): File {
+    val buildDir = project.buildDir.toString()
+    var outputDir = variantOutput.dirName
+    if (variantOutput.dirName.endsWith("dpi" + File.separator)) {
+        outputDir = File(variantOutput.dirName).parent
+        if (outputDir == null) { // if only density splits enabled
+            outputDir = ""
+        }
+    }
+    return Paths.get(buildDir, *path, variant.dirName, outputDir, "mapping.txt").toFile()
+}
+
+/**
+ * Returns true if the DexGuard plugin has been applied to the project
+ */
+internal fun Project.hasDexguardPlugin(): Boolean {
+    return pluginManager.hasPlugin("dexguard")
+}
+
+/**
+ * Retrieves the major version of DexGuard in use in the project
+ */
+internal fun getDexguardMajorVersionInt(project: Project): Int {
+    val version = GroovyCompat.getDexguardVersionString(project) ?: return -1
+    val major = version[0]
```

Comment:
Won't this break when DexGuard 10 is released? It might be safer to use `VersionNumber` to parse this, like we do elsewhere

Reply:
Good call - I've updated the code to use `VersionNumber` for parsing this.